modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -343,12 +387,22 @@ else if (Objects.equals(BinaryExpression.Operation.NE, operation)) {
 	}
 
 	private Object _getValue(Object left, Object right) {
+		EntityField entityField = _getEntityField(left);
+
+		EntityField.Type entityType = entityField.getType();
+
+		if (entityType.equals(EntityField.Type.DATE_TIME) &&
+			StringUtil.startsWith(_getDriverName(), "HSQL")) {
+
```

Comment:
@thienquach87, you may simplify the code fragment and replace StringUtil.startsWith() with the following code
```
DB db = DBManagerUtil.getDB();
if (db.getDBType() == DBType.HYPERSONIC) {}
```

Reply:
thank you, I updated source based on your suggestion